### PR TITLE
framework(engine): remove 'JobMasterID' method from 'BaseJobMaster' interface

### LINKED ISSUE #6615: engine: Remove the unnecessary method 'JobMasterID' in JobMaster interface
### Is your feature request related to a problem?

No

### Describe the feature you'd like

Although 'JobMaster' is a combination of 'Master' and 'Slave' inside, for the user, 'JobMaster' is an independent  character which has the ability to manager its 'workers'. 

So if we offer both the 'JobMasterID' and 'ID' interface method in  'JobMaster', it will confuses user.
The normal doubt Is : 

1. What's the difference of  JobMasterID and ID? 
2. What do they mean respectively?
3. Are they equal?

So it's better to remove the unnecessary method 'JobMasterID' in JobMaster interface to avoid confusion.

### Describe alternatives you've considered

Remove the unnecessary method 'JobMasterID' in JobMaster interface

### Teachability, Documentation, Adoption, Migration Strategy

_No response_

### PR DESCRIPTION
**Pull request by maxshuang**
### What problem does this PR solve?

Issue Number: close #6615 

### What is changed and how it works?
'JobMasterID' and 'ID' return the same id to user and we don't need the `JobMasterID` to return `JobManagerID` actually,
so remove 'JobMasterID'  method to avoid confusion

### Release note 

```release-note
 `None`.
```

### CODE REVIEW

**Review by GMHDBJD — APPROVED** [review]
LGTM

File: engine/framework/base_jobmaster.go [hunk 1]
```diff
@@ -275,13 +273,10 @@ func (d *DefaultBaseJobMaster) Workload() model.RescUnit {
 	return d.worker.Workload()
 }
 
-// ID delegates the ID of inner worker
-func (d *DefaultBaseJobMaster) ID() runtime.RunnableID {
-	return d.worker.ID()
-}
```
Comment (amyangfei) [inline engine/framework/base_jobmaster.go:281]: This function signature should not be changed.

**Review by amyangfei — APPROVED** [review]

File: engine/framework/base_jobmaster.go (hunk 1 above)
Comment (maxshuang) [reply engine/framework/base_jobmaster.go]: Yes!! You are right. Fix it.
BTW, I think it's strange for the `ID()` of `Worker` to return a runtime-related type `runtime.RunnableID`, but not a `WorkerID`.

**Comment by maxshuang** [comment]
@GMHDBJD Could you please help fix the unit test error in `tiflow/engine/jobmaster/dm`?  Just remove the meaningless ID in your cases.

**Comment by maxshuang** [comment]
Hold this pr untill https://github.com/pingcap/tiflow/pull/6620 being merged to master.
#6620 will remove the usage of `JobMasterID` method of dm.

**Pull request merged**